### DISCUSSION TITLE
(firefox) prevent full installer from sending Install Ping

### DISCUSSION THREAD

**Discussion by msoxzw** [chocolatey-community/chocolatey-packages#1927] [Ideas]
In order to protect privacy, adding ["/LaunchedFromStub"](https://searchfox.org/mozilla-central/rev/26a6a38fb515dbab0bb459c40ec4b877477eefef/browser/installer/windows/nsis/installer.nsi#1039-1044) command line option to the Firefox full installer could prevent [Install Ping](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/install-ping.html) from being sent.

**Comment by TheCakeIsNaOH** [comment]
Looking at the linked `.nsi` script, the `/LaunchedFromStub` argument is referenced a number of times, and appears to change more behavior than purely not sending an Install Ping. Additionally, I do not think there would be a guarantee that there may not be changes in how this option behaves in future versions. 

Therefore, I do not think this should be added as a default option passed to the installer. But potentially I think it would be ok to add it gated behind a package parameter so only passed to the installer if the user specifically wants it. 

Also, individual users can also use `--install-arguments` to pass it directly to the installer, without requiring changes to the package.

**Reply by msoxzw** [reply]
Thanks for digging for `/LaunchedFromStub` usage in the script. The `/LaunchedFromStub` arguments in the other places are primarily used for telemetry, and I have used it over one year without any issues. Nevertheless, this undocumented argument should be used prudently. Alternatively, the most privacy-preserving way of installing Firefox, I believe, is to extract the installer by [`/ExtractDir=[directory]`](https://firefox-source-docs.mozilla.org/browser/installer/windows/installer/FullConfig.html) and execute [`/PostUpdate`](https://firefox-source-docs.mozilla.org/browser/installer/windows/installer/Helper.html#postupdate), so none of telemetry data will be generated.

**Reply by AdmiringWorm** [reply]
Considering that the `/LaunchedFromStub` argument is completely undocumented, I am a bit hesitant to add such a thing as a package parameter. Since it is undocumented, it can change at any time, but we would be stuck with trying to support the parameter. If it gets documented at some point, we can certainly add this as a package parameter.

> Alternatively, the most privacy-preserving way of installing Firefox, I believe, is to extract the installer by [/ExtractDir=[directory]](https://firefox-source-docs.mozilla.org/browser/installer/windows/installer/FullConfig.html) and execute [/PostUpdate](https://firefox-source-docs.mozilla.org/browser/installer/windows/installer/Helper.html#postupdate), so none of telemetry data will be generated.

I don't believe this is something that we will ever add to this package, at the very least, not anytime soon.
